Firestore/core/src/firebase/firestore/local/local_serializer.cc

```diff
@@ -67,30 +70,29 @@ firestore_client_MaybeDocument LocalSerializer::EncodeMaybeDocument(
     case MaybeDocument::Type::Document:
       result.which_document_type = firestore_client_MaybeDocument_document_tag;
       result.document = EncodeDocument(static_cast<const Document&>(maybe_doc));
-      // TODO(rsgowman): heldwriteacks:
-      // result.has_committed_mutations = existing_doc.HasCommittedMutations();
+      result.has_committed_mutations =
+          static_cast<const Document&>(maybe_doc).has_committed_mutations();
       return result;
 
     case MaybeDocument::Type::NoDocument:
       result.which_document_type =
           firestore_client_MaybeDocument_no_document_tag;
       result.no_document =
           EncodeNoDocument(static_cast<const NoDocument&>(maybe_doc));
-      // TODO(rsgowman): heldwriteacks:
-      // result.has_committed_mutations = no_doc.HasCommittedMutations();
+      result.has_committed_mutations =
+          static_cast<const NoDocument&>(maybe_doc).has_committed_mutations();
       return result;
 
     case MaybeDocument::Type::UnknownDocument:
       result.which_document_type =
           firestore_client_MaybeDocument_unknown_document_tag;
       result.unknown_document =
```

Comment:
Please double-check with Gil, but I think that the intended way of downcasting documents is to just construct the derived type by passing over the base type to the constructor, like Objective-C does it:
```cpp
    UnknownDocument unknownDocument(document);
```
Note that it's cheap to construct. This will also avoid having to repeat the cast twice (in the cases above).

Reply:
Nit: The braces around this case now seem unnecessary.